.github/workflows/tmp_post_create_action.yaml

```diff
@@ -0,0 +1,61 @@
+name: Post-Create Setup
+
+on:
+  push:
+    branches:
+      - main  # Ensure this matches the default branch of your template
+
+jobs:
+  setupRepo:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Set up Git
+        run: |
+          git config --global user.name "${{ github.actor }}"
+          git config --global user.email "${{ github.actor }}@users.noreply.github.com"
+
+      - name: Replace {APP-NAME} with repository name
+        run: |
+          repo_name=${{ github.event.repository.name }}
+          grep -rl '{APP-NAME}' . | xargs sed -i "s/{APP-NAME}/$repo_name/g"
```

Comment:
I am using the repo name, but providing the general metadata in a different format may be better. Open for ideas